mantidproject/mantid
# AlgorithmSelector Widget to Observe AlgorithmFactory in Workbench

**Description of work.**
This PR introduces a `AlgorithmFactoryObserver`, in the style of `AnalysisDataServiceObserver`, to make the AlgorithmSelector widget in workbench refresh if a script adds a new algorithm

**To test:**
In workbench, run the script outlined [here](https://www.mantidproject.org/Extending_Mantid_With_Python:_Exercise_2_Solution).
In the AlgorithmSelector widget, check that FibonacciExercise exists and can be run

Fixes #25145 

*This does not require release notes* because **fix to issue in workbench - not yet released**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
